lib/ask/runtime/questionnaire_simulator.ex

```diff
@@ -159,20 +159,28 @@ defmodule Ask.Simulation.SubmittedStep do
   alias Ask.Runtime.Reply
   alias Ask.Questionnaire
 
-  def build_from(reply, questionnaire) do
-    responses = Reply.stores(reply)
-                |> Enum.map(fn {step_name, value} ->
-                  # find function is used since there is a restriction that two steps cannot have the same store variable name
-                  %{"id" => id} = questionnaire |> Questionnaire.all_steps |> Enum.find(fn step -> step["store"] == step_name end)
-                  %{response: value, step_id: id}
+  # Extract new responses from respondent
+  def build_from_responses(respondent, %{submissions: submissions, questionnaire: quiz}) do
+    present_submissions = submissions |> Enum.map(fn submitted_step -> submitted_step.step_name end)
+    new_responses = respondent.responses |> Enum.filter(fn response -> not (present_submissions |> Enum.member?(response.field_name)) end)
+
+    new_responses
+    |> Enum.map(fn %{field_name: step_name, value: value} ->
+      # find function is used since there is a restriction that two steps cannot have the same store variable name
+      %{"id" => id} = quiz |> Questionnaire.all_steps |> Enum.find(fn step -> step["store"] == step_name end)
+      %{response: value, step_id: id, step_name: step_name}
     end)
 
-    explanation_steps = Reply.steps(reply)
-                        |> Enum.filter(fn step -> step.type == "explanation" end)
-                        |> Enum.filter(fn step -> step.title not in ["Thank you", "Error"] end)
-                        |> Enum.map(fn step -> %{step_id: step.id} end)
-    responses ++ explanation_steps
   end
+
+  # Explanation submitted-steps must be extracted from reply since aren't stored as responses
+  def new_explanations(reply) do
+    Reply.steps(reply)
+    |> Enum.filter(fn step -> step.type == "explanation" end)
+    |> Enum.filter(fn step -> step.title not in ["Thank you", "Error"] end)
```

Comment:
Those are the titles surveda puts to this steps when creates them based on the questionnaire settings `"than_you_message"` and `"error_message"`. All this happens in `flow.ex` (check lines [364](https://github.com/instedd/surveda/blob/2d58ba9441ea7a8eacc5043a0b7ce9a674c1595a/lib/ask/runtime/flow.ex#L364) and [256](https://github.com/instedd/surveda/blob/2d58ba9441ea7a8eacc5043a0b7ce9a674c1595a/lib/ask/runtime/flow.ex#L256))

This steps are ignored in the `submissions` field since are not _real_ steps (user don't configure them as steps)